compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -5941,6 +5997,8 @@ object Types extends TypeUtils {
         samClass(tp.underlying)
       case tp: AnnotatedType =>
         samClass(tp.underlying)
+      case tp: FlexibleType =>
+        samClass(tp.superType)
```

Comment:
Should ideally be `tp.underlying`, like the other cases. There are technical reasons why it's supertype for `AppliedType`, but these don't hold for FlexibleType.